packages/apple-pay-integration/src/apple-pay-payment-strategy.ts

```diff
@@ -248,4 +267,31 @@ export default class ApplePayPaymentStrategy implements PaymentStrategy {
             );
         }
     }
+
+    private async _getBraintreeDeviceData() {
+        const data = await this._braintreeIntegrationService.getDataCollector();
+
+        return data.deviceData;
+    }
+
+    private async _initializeBraintreeIntegrationService() {
+        const state = await this._paymentIntegrationService.loadPaymentMethod(
```

Comment:
I believe you can't avoid `loadPaymentMethod()` call by passing state or needed values from it as arguments

Reply:
Here I need to load the payment method again because I need to get the **clientToken** to initialize the **braintreeIntegrationService**

Reply:
you can pass storeConfig and braintreePaymentMethod from this place https://github.com/bigcommerce/checkout-sdk-js/pull/2165/files#diff-fe2a47383384ef438fc5b63603ec00c11e23b8e5289b37a1fa8ced442eb6c2ffR71 .
You already have state here https://github.com/bigcommerce/checkout-sdk-js/pull/2165/files#diff-fe2a47383384ef438fc5b63603ec00c11e23b8e5289b37a1fa8ced442eb6c2ffR66

Reply:
resolved